### PR TITLE
Update `UP032` to autofix multi-line triple-quoted string

### LINKED ISSUE #5854: Extend `UP032` (`f-string`) to support auto-fixing multi-line triple-quoted strings
UP032 currently ignores multi-lines strings:

https://github.com/astral-sh/ruff/blob/a4e5e3205fce46f00456d7a7b015bd1140a7abe8/crates/ruff/src/rules/pyupgrade/rules/f_strings.rs#L328-L331

## Examples

```python
# Ignored
a = (
  "{}"
  "{}".format(1, 2)
)

# Ignored
a = """
{}
""".format(1)

# Not ignored
a = "{}".format(1)
```

Is it possible to support auto-fixing the second case?

### PR DESCRIPTION
**Pull request by harupy**
## Summary

Resolve #5854

## Test Plan

New test cases

### CODE REVIEW

File: crates/ruff/src/rules/pyupgrade/rules/f_strings.rs [hunk 1]
```diff
@@ -338,7 +338,10 @@ pub(crate) fn f_strings(
 
     // Avoid refactors that exceed the line length limit.
     let col_offset = template.start() - checker.locator.line_start(template.start());
-    if col_offset.to_usize() + contents.len() > line_length.get() {
+    if contents
+        .split('\n')
+        .any(|line| col_offset.to_usize() + line.len() > line_length.get())
```
Comment (konstin) [inline crates/ruff/src/rules/pyupgrade/rules/f_strings.rs:343]: The `col_offset` is only correct for the first line, all other lines have an offset of zero.
```python
aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa = """
{}
{}
{}
""".format(
1,
2,
333333333333333333333,
)
```
In this example we don't suggest the autofix even though we're inside the line limit. This is a false negative for autofixing though, so i'm also fine with leaving a TODO comment and merging

**Review by konstin — APPROVED** [review]

File: crates/ruff/src/rules/pyupgrade/rules/f_strings.rs [hunk 2]
```diff
@@ -338,7 +338,10 @@ pub(crate) fn f_strings(
 
     // Avoid refactors that exceed the line length limit.
     let col_offset = template.start() - checker.locator.line_start(template.start());
-    if col_offset.to_usize() + contents.len() > line_length.get() {
+    if contents
+        .split('\n')
```
[the remark this answers is not in the document]
Comment (konstin) [reply crates/ruff/src/rules/pyupgrade/rules/f_strings.rs:342]: good point, yep

File: crates/ruff/src/rules/pyupgrade/rules/f_strings.rs (hunk 1 above)
Comment (harupy) [reply crates/ruff/src/rules/pyupgrade/rules/f_strings.rs:343]: Can we only use `col_offset` for the first line, and ignore it for the rest?

```diff
diff --git a/crates/ruff/src/rules/pyupgrade/rules/f_strings.rs b/crates/ruff/src/rules/pyupgrade/rules/f_strings.rs
index d79dbd560..695b3c35a 100644
--- a/crates/ruff/src/rules/pyupgrade/rules/f_strings.rs
+++ b/crates/ruff/src/rules/pyupgrade/rules/f_strings.rs
@@ -338,10 +338,10 @@ pub(crate) fn f_strings(
 
     // Avoid refactors that exceed the line length limit.
     let col_offset = template.start() - checker.locator.line_start(template.start());
-    if contents
-        .lines()
-        .any(|line| col_offset.to_usize() + line.len() > line_length.get())
-    {
+    if contents.lines().enumerate().any(|(idx, line)| {
+        let offset = if idx == 0 { col_offset.to_usize() } else { 0 };
+        offset + line.len() > line_length.get()
+    }) {
         return;
     }
```

Comment (konstin) [reply crates/ruff/src/rules/pyupgrade/rules/f_strings.rs:343]: yep! this should get a comment tough explaining why we do this

File: crates/ruff/src/rules/pyupgrade/rules/f_strings.rs [hunk 3]
```diff
@@ -338,10 +338,17 @@ pub(crate) fn f_strings(
...
-        .any(|line| col_offset.to_usize() + line.len() > line_length.get())
-    {
+    if contents.lines().enumerate().any(|(idx, line)| {
+        // If `template` is a multiline string, `col_offset` should only be applied to the first
+        // line. See this example:
+        // ```
+        // a = """{}        -> offset = col_offset (= 4)
+        // {}               -> offset = 0
+        // """.format(0, 1) -> offset = 0
+        // ```
```
Comment (konstin) [inline crates/ruff/src/rules/pyupgrade/rules/f_strings.rs:348]: great comment, thanks!

**Pull request merged**